src/controllers/super-admins.js

```diff
@@ -1,115 +1,161 @@
-      res.send(error || ('Super admin added'));
+// GET ALL SUPERADMINS
+const getAllSuperadmins = async (req, res) => {
+  try {
+    const allSuperadmins = await models.SuperAdmin.find({});
+    return res.status(200).json({
+      message: 'All Super-Admins',
+      data: allSuperadmins,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: error,
+      data: {},
+      error: true,
     });
   }
-});
+};
 
-// delete super admin
-router.delete('/delete/:id', (req, res) => {
-  const supAdmId = Number(req.params.id);
-  const filteredUsers = superAdmins.filter((user) => user.id !== supAdmId);
-  if (superAdmins.length === filteredUsers.length) {
-    res.status(400);
-    res.send(`could not delete because super admin with id "${supAdmId}" does not exist`);
-  } else {
-    fs.writeFile('src/data/super-admins.json', JSON.stringify(filteredUsers), (error) => {
-      res.send(error || ('Super admin deleted'));
+// CREATE A NEW SUPERADMIN
+const createSuperadmin = async (req, res) => {
+  try {
+    const {
+      firstName, lastName, email, password, active,
+    } = req.body;
+    const newSuperadmin = new models.SuperAdmin({
+      firstName,
+      lastName,
+      email,
+      password,
+      active,
+    });
+    const result = await newSuperadmin.save();
+    return res.status(201).json({
+      message: 'Super-Admin created',
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: error,
+      data: {},
+      error: true,
     });
   }
-});
+};
 
-// edit super admin
-router.put('/edit/:id', (req, res) => {
-  const supAdmId = Number(req.params.id);
-  const newArray = superAdmins.find((superAdmin) => superAdmin.id === supAdmId);
-  const filteredArray = superAdmins.filter((superAdmin) => superAdmin.id !== supAdmId);
-  if (newArray) {
-    const updatedSA = {
-      id: supAdmId,
-      firstName: (req.body.firstName || newArray.firstName),
-      lastName: (req.body.lastName || newArray.lastName),
-      email: (req.body.email || newArray.email),
-      password: (req.body.password || newArray.password),
-      active: (req.body.active ?? newArray.active),
-    };
-    filteredArray.push(updatedSA);
-    fs.writeFileSync('src/data/super-admins.json', JSON.stringify(filteredArray));
-    res.send(`Super admin with id "${supAdmId}" edited`);
-  } else {
-  //  res.status(400);
-    res.send(`Could not edit because super admin with id "${supAdmId}" does not exist`);
+// SEARCH SUPERADMINS WITH FILTERS INCLUDED IN BODY
+const getFilteredSuperadmins = async (req, res) => {
+  try {
+    const {
+      firstName, lastName, email, active,
+    } = req.body;
+    const stringQuery = await models.SuperAdmin.find({
+      firstName: { $regex: new RegExp(firstName || '', 'i') },
+      lastName: { $regex: new RegExp(lastName || '', 'i') },
+      email: { $regex: new RegExp(email || '', 'i') },
+    });
```

Comment:
```suggestion
    const stringQuery = await models.SuperAdmin.find({
      firstName: { $regex: new RegExp(firstName || '', 'i') },
      lastName: { $regex: new RegExp(lastName || '', 'i') },
      email: { $regex: new RegExp(email || '', 'i') },
    });
```
por que usamos el RegExp? lo pienso desde el lado de que si estoy buscando algo, y lo ingreso mal, no voy a encontrar nada, por que validaria un get?

Reply:
Respecto al new RegExp, es una forma de hacer que pasandole solo una parte del string te devuelva true si esta incluido (todo esto dentro de las query de mongodb). Cuando hice los test, le pasa, por ejemplo, solo una vocal, y te tira todos los nombres que tienen esa vocal (es cierto que no hice validacion de numeros...) Lo mismo con el mail, por ejemplo si le paso un @, como todos los mail tienen arroba, me devuelve todos los usuarios. por otro lado, si es un string vacio (o si no le pase algun campo en el body), el regex de"" da siempre true, o sea que es lo mismo a que no este.
la i como segundo parametro es para hacerlo case insensitive